src/xinterpreter.cpp

```diff
@@ -498,70 +322,19 @@ namespace xcpp
         return s;
     }
 
-    static int printf_jit(const char* format, ...)
-    {
-        std::va_list args;
-        va_start(args, format);
-
-        std::string buf = c_format(format, args);
-        std::cout << buf;
-
-        va_end(args);
-
-        return buf.size();
-    }
-
-    static int fprintf_jit(std::FILE* stream, const char* format, ...)
-    {
-        std::va_list args;
-        va_start(args, format);
-
-        int ret;
-        if (stream == stdout || stream == stderr)
-        {
-            std::string buf = c_format(format, args);
-            if (stream == stdout)
-            {
-                std::cout << buf;
-            }
-            else if (stream == stderr)
-            {
-                std::cerr << buf;
-            }
-            ret = buf.size();
-        }
-        else
-        {
-            // Just forward to vfprintf.
-            ret = vfprintf(stream, format, args);
-        }
-
-        va_end(args);
-
-        return ret;
-    }
-
     void interpreter::redirect_output()
     {
         p_cout_strbuf = std::cout.rdbuf();
         p_cerr_strbuf = std::cerr.rdbuf();
 
         std::cout.rdbuf(&m_cout_buffer);
         std::cerr.rdbuf(&m_cerr_buffer);
-
-        // Inject versions of printf and fprintf that output to std::cout
-        // and std::cerr (see implementation above).
-        inject_symbol("printf", llvm::pointerToJITTargetAddress(printf_jit), *m_interpreter);
-        inject_symbol("fprintf", llvm::pointerToJITTargetAddress(fprintf_jit), *m_interpreter);
```

Comment:
Just curious to know if/why we don't need these functions anymore ?

Reply:
We do have a better way to redirect i/o streams with CppInterOp. That will be part of a new PR as we discussed to do a minimal change with this one.

Reply:
Where does std::clog writes to? We only redirect file descriptors 1 and 2.

Reply:
std::clog will print to stdout by default - so will be directed to the frontend.

Reply:
Isn’t that what we should expect if a user type std::clog in a jupyter cell? If not we can make the redirection of std::clog go somewhere else.